spring-cloud-azure-autoconfigure/src/main/java/com/microsoft/azure/spring/cloud/autoconfigure/telemetry/TelemetryTracker.java

```diff
@@ -42,11 +52,27 @@ public TelemetryTracker(Azure azure, String resourceGroup) {
         this.defaultProperties.put(PROPERTY_INSTALLATION_ID, TelemetryUtils.getHashMac());
     }
 
+    private TelemetryClient getTelemetryClient(TelemetryProperties telemetryProperties) {
+        final TelemetryClient client = new TelemetryClient();
+        final String instrumentationKey = telemetryProperties.getInstrumentationKey();
+
+        if (StringUtils.hasText(instrumentationKey) && instrumentationKey.length() == INSTRUMENTATION_KEY_LENGTH) {
+            client.getContext().setInstrumentationKey(instrumentationKey);
+        } else {
+            client.getContext().setInstrumentationKey(TELEMETRY_INVALID_KEY);
+            LOG.warn("Telemetry instrumentationKey {} is invalid", instrumentationKey);
+        }
+
+        return client;
+    }
+
     private void trackEvent(@NonNull String name, @NonNull Map<String, String> customProperties) {
-        this.defaultProperties.forEach(customProperties::putIfAbsent);
+        if (!this.client.getContext().getInstrumentationKey().equals(TELEMETRY_INVALID_KEY)) {
```

Comment:
What does this mean?

Reply:
If there key is invalid, we may not send the trigger. And it will be set in constructor depends on key values.